protocols/relay/src/behaviour.rs

```diff
@@ -271,6 +286,78 @@ impl Behaviour {
             circuits: Default::default(),
             queued_actions: Default::default(),
             external_addresses: Default::default(),
+            status: Status::Disable,
+            auto_status_change: true,
+            waker: None,
+        }
+    }
+
+    pub fn set_status(&mut self, status: Option<Status>) {
+        match status {
+            Some(status) => {
+                self.status = status;
+                self.auto_status_change = false;
+                self.reconfigure_relay_status();
+            }
+            None => {
+                self.auto_status_change = true;
+                self.determine_relay_status_from_external_address();
+            }
+        }
+        if let Some(waker) = self.waker.take() {
+            waker.wake();
+        }
+    }
+
+    fn reconfigure_relay_status(&mut self) {
+        if self.reservations.is_empty() {
+            return;
+        }
+
+        for (peer_id, connections) in self.reservations.iter() {
+            self.queued_actions
+                .extend(connections.iter().map(|id| ToSwarm::NotifyHandler {
+                    peer_id: *peer_id,
+                    handler: NotifyHandler::One(*id),
+                    event: Either::Left(handler::In::SetStatus {
+                        status: self.status,
+                    }),
+                }));
+        }
```

Comment:
Don't we need to inform all handlers? I think otherwise existing connection handlers without an active reservation would still have the outdated status.